### ISSUE TITLE
 Filter results based on Geo-location field

### ISSUE THREAD

**Issue by nicksav** [oramasearch/orama#212]
**Is your feature request related to a problem? Please describe.**
Adding support to filter based on location. 

**Describe the solution you'd like**
As a minimum provide ability to filter results inside a rectangle - boundingBox

**Additional context**
I think filtering a key here. Yep, search is good, but filters, facets, geo search is highly required.

**Comment by micheleriva** [comment]
Hi @nicksav,
that would be a nice feature. Let me ping as soon as we assign it a priority in our roadmap.

Thanks!

**Comment by sreenath-n** [comment]
@micheleriva these two features are also very useful in most of the modern apps:
1. Filter based on center geopoint and radius (Eg. to find restaurants nearby)
2. Sort results based on distance from a geopoint (Eg. to find restaurants nearby and sort based on nearest first)

Example from `Typesense`:
```
'filter_by' : 'location:(48.90615915923891, 2.3435897727061175, 5.1 km)',
'sort_by'   : 'location(48.853, 2.344):asc'
```
`Typesense` Doc: https://typesense.org/docs/0.23.1/api/geosearch.html

**Comment by W7T2A** [comment]
+1 for this, is there any update in prio?

**Comment by sreenath-n** [comment]
This is the only blocker for us to switch from Typesense to Orama Search.

**Comment by imlautaro** [comment]
+1 for this too

**Comment by micheleriva** [comment]
Hi y'all. Still to be prioritized, will update this issue as soon as we prioritize it!

**Comment by Leanvitale** [comment]
We could also use this update, since it is the only thing that prevents us from implementing Orama in our Startup.

**Comment by micheleriva** [comment]
We're on it

**Comment by micheleriva** [comment]
This feature is available for preview in `v2.0.0-beta.1`. Has been implemented with https://github.com/oramasearch/orama/pull/505. Please try it and give us some feedback! 🙂
